src/clients/cache/momento/commands/set_if_absent.rs

```diff
@@ -8,15 +9,33 @@ pub async fn set_if_absent(
 ) -> std::result::Result<(), ResponseError> {
     SET_IF_ABSENT.increment();
 
-    let result = timeout(
+    match timeout(
         config.client().unwrap().request_timeout(),
         client.set_if_absent(
             cache_name,
             (*request.key).to_owned(),
             (*request.value).to_owned(),
         ),
     )
-    .await;
-
-    record_result!(result, SET_IF_ABSENT)
+    .await
```

Comment:
I wonder if we could move some of this boilerplate code into `record_result!`.  Especially for conditional write responses where we do not care about the actual value of the item in the response and just want to record metrics for stored vs not_stored. The macro already seems to have some logic based on [3 arguments being passed to it](https://github.com/iopsystems/rpc-perf/blob/b5a7a65bccf09ef5c9cdf528bf0167afda71aa4d/src/clients/cache/momento/commands/set.rs#L26)

Reply:
I think you could, but it'd need enough other code to support inspecting arbitrary enums to check stored/not_stored (some sort of trait that's implemented for each conditional write response enum?) that I don't know if it'd save any real code. 

record_result!() could maybe still be used here for the EX and TIMEOUT - it'd expect to increment something like SET_IF_ABSENT_OK - which should exist since the `request!` macro does some `paste` magic to generate _OK, _EX, and _TIMEOUT metrics for the same base request type. Then this could be a record_result!() and an 
```
record_result!(...);

if let Ok(Ok(r)) = r { 
    match r { 
        /// enum variant handling 
    }
 }
 ```
 
 Might be worth making that change